src/demo/service/demo-mission-runtime.service.ts

```diff
@@ -1,24 +1,22 @@
 import { Injectable } from '@angular/core';
-import { Headers, Http, RequestOptions, Response } from '@angular/http';
-import { Observable, Subject, BehaviorSubject } from 'rxjs';
 
-import { MissionRuntimeService, Cluster } from '../../app/launcher/launcher.module';
-import { Mission } from '../../app/launcher/launcher.module';
-import { Runtime } from '../../app/launcher/launcher.module';
+import { MissionRuntimeService } from '../../app/launcher/launcher.module';
+import { Observable} from 'rxjs/Observable';
+import { Catalog } from '../../app/launcher/model/catalog.model';
 
-import { HelperService } from '../../app/launcher/service/helper.service';
-import { TokenProvider } from '../../app/service/token-provider';
+const osioMockData = require('../../assets/mock/demo-catalog-osio.json') as Catalog;
+const launchMockData = require('../../assets/mock/demo-catalog-launch.json') as Catalog;
 
-const MockData = require('../../assets/mock/demo-mission-runtime.json')
 
 @Injectable()
-export class DemoMissionRuntimeService implements MissionRuntimeService {
+export class DemoMissionRuntimeService extends MissionRuntimeService {
 
-  getMissions(): Observable<Mission[]> {
-    return Observable.of(MockData.missions);
+  getCatalog(): Observable<Catalog> {
+    const isLauncher = location.href.indexOf('launcher') !== -1;
```

Comment:
@edewit I only found this dirty way of finding if it's launcher or not in the demo. The origin config is not dynamic and can't be used.. do you have a better idea?

Reply:
for the demo app this is fine IMO